aiohttp/client.py

```diff
@@ -785,11 +775,15 @@ async def _ws_connect(
             extstr = ws_ext_gen(compress=compress)
             real_headers[hdrs.SEC_WEBSOCKET_EXTENSIONS] = extstr
 
-        if not isinstance(ssl, SSL_ALLOWED_TYPES):
-            raise TypeError(
-                "ssl should be SSLContext, bool, Fingerprint, "
-                "or None, got {!r} instead.".format(ssl)
+        # For the sake of backward compatibility, if user passes in None, convert it to True
+        if ssl is None:
+            warnings.warn(
+                "None is a deprecated value for the ssl parameter; "
+                "using True for default ssl context.",
             )
```

Comment:
Deprecation warnings should set the `DeprecationWarning` (or `FutureDeprecationWarning`) type + usually `stacklevel=2`.

Reply:
This will be relevant for the backport, yes.
But, this code shouldn't exist in master, where None should not be an allowed value anymore.

Reply:
I'd argue that a PR deprecating it should go in separately, followed by a non-backported PR that actually does the removal.

Reply:
Hmm, the deprecation is likely to create more changes though, which then need to be reverted in the removal. So, that seems like more work to me.

Reply:
Actually, maybe not, I'm thinking of some changes that are no longer relevant. I don't mind which way it's done.